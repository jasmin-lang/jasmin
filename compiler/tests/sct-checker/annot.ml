open Jasmin
open Common
open Sct_checker_forward

let () =
  let p = load_file "annot.jazz" in
  ty_prog p [] |> List.iter (Format.printf "%a@." pp_funty)
