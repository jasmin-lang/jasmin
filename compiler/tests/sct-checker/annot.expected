modmsf f : #transient ->
#secret
output corruption: #public
 constraints:


modmsf id : #poly = { n = k, s = k} ->
#poly = { n = k, s = k}
output corruption: #public
 constraints:
k <= k, 

nomodmsf id1 : #poly = { n = ?17, s = ?16} ->
#poly = { n = ?17, s = ?16}
output corruption: #public
 constraints:


nomodmsf id2 : #[ptr = { n = ?27, s = ?26}, val = { n = ?25, s = ?24}] ->
#[ptr = { n = ?27, s = ?26}, val = { n = ?25, s = ?24}]
output corruption: #public
 constraints:


nomodmsf id3 : #public ->
#public
output corruption: #public
 constraints:


nomodmsf id4 : #transient ->
#transient
output corruption: #public
 constraints:


nomodmsf id5 : #secret ->
#secret
output corruption: #public
 constraints:


